actors/builtin/reward/reward_actor.go

```diff
@@ -67,10 +66,14 @@ type AwardBlockRewardParams struct {
 // - a penalty amount, provided as a parameter, which is burnt,
 func (a Actor) AwardBlockReward(rt vmr.Runtime, params *AwardBlockRewardParams) *adt.EmptyValue {
 	rt.ValidateImmediateCallerIs(builtin.SystemActorAddr)
-	AssertMsg(params.Miner.Protocol() == addr.ID,
-		"miner address must be ID-address protocol (%d), was protocol %d", addr.ID, params.Miner.Protocol())
 	AssertMsg(rt.CurrentBalance().GreaterThanEqual(params.GasReward),
 		"actor current balance %v insufficient to pay gas reward %v", rt.CurrentBalance(), params.GasReward)
+
+	miner, ok := rt.ResolveAddress(params.Miner)
+	if !ok {
+		rt.Abortf(exitcode.ErrIllegalState, "failed to resolve given owner address")
+	}
```

Comment:
The sender here will be the VM/interpreter, which has no excuse not to provide an ID address (which it gets from the block header). So I prefer the assertion, but this is ok.